src/core/thread/link_quality.cpp

```diff
@@ -154,7 +154,10 @@ void LinkQualityInfo::AddRss(int8_t aNoiseFloor, int8_t aRss)
 {
     uint8_t oldLinkQuality = kNoLinkQuality;
 
-    mLastRss = aRss;
+    if (aRss != OT_RADIO_RSSI_INVALID)
```

Comment:
Should we make this `VerifyOrExit()` and just jump to `exit:` on invalid rssi value?

Reply:
Sure, the averager already ignores it, so adding in a VerifyOrExit shouldn't change the logic flow anyways.